src/collapse/perf.rs

```diff
@@ -1,114 +1,169 @@
+    }
+}
 
-    /// Annotate JIT functions with a `_[j]` suffix.
+/// Perf folder configuration options.
+#[derive(Clone, Debug)]
+pub struct Options {
+    /// Annotate JIT functions with a `_[j]` suffix. Default is `false`.
     pub annotate_jit: bool,
 
-    /// Annotate kernel functions with a `_[k]` suffix.
+    /// Annotate kernel functions with a `_[k]` suffix. Default is `false`.
     pub annotate_kernel: bool,
 
-    /// Only consider samples of the given event type (see `perf list`).
-    ///
-    /// If this option is set to `None`, it will be set to the first encountered event type.
+    /// Only consider samples of the given event type (see `perf list`). If this option is
+    /// set to `None`, it will be set to the first encountered event type. Default is `None`.
     pub event_filter: Option<String>,
-}
 
-#[derive(Copy, Clone, Debug)]
-enum EventFilterState {
-    None,
-    Defaulted,
-    Warned,
+    /// Include raw addresses (e.g., `0xbfff0836`) where symbols can't be found.
+    /// Default is `false`.
+    pub include_addrs: bool,
+
+    /// Include PID in the root frame. If disabled, the root frame is given the name of the
+    /// profiled process. Default is `false`.
+    pub include_pid: bool,
+
+    /// Include TID and PID in the root frame. Implies `include_pid`. Default is `false`.
+    pub include_tid: bool,
+
+    /// The number of threads to use. Default is the number of logical cores on your machine.
+    pub nthreads: usize,
 }
 
-impl Default for EventFilterState {
+impl Default for Options {
     fn default() -> Self {
-        EventFilterState::None
+        Self {
+            annotate_jit: false,
+            annotate_kernel: false,
+            event_filter: None,
+            include_addrs: false,
+            include_pid: false,
+            include_tid: false,
+            nthreads: *collapse::DEFAULT_NTHREADS,
+        }
     }
 }
 
 /// A stack collapser for the output of `perf script`.
 ///
 /// To construct one, either use `perf::Folder::default()` or create an [`Options`] and use
 /// `perf::Folder::from(options)`.
-#[derive(Default)]
 pub struct Folder {
-    /// All lines until the next empty line are stack lines.
-    in_event: bool,
+    // State...
+    /// General String cache that can be used while processing lines. Currently only used to keep
+    /// track of functions for Java inlining.
+    cache_line: Vec<String>,
 
-    /// Skip all stack lines in this event.
-    skip_stack: bool,
+    /// Similar to, but different from, the `event_filter` field on `Options`
+    ///
+    /// * Field on `Options` represents user's provided configuration and will never change.
+    /// * This field, on the other hand, although initially set to the user's provided
+    ///   configuration, represents state that may change as data is processed. In particular, if
+    ///   the user provided `None` for his/her initial configuration, this field **will** change to
```

Comment:
```suggestion
    ///   the user provided `None` for their initial configuration, this field **will** change to
```

Reply:
Grammatically controversial (**[link](https://en.wikipedia.org/wiki/Singular_they)**) - I like it! :)